src/mesh/index_derivs.cxx

```diff
@@ -2852,12 +2957,21 @@ const Field2D Mesh::indexFDDY(const Field2D &v, const Field2D &f, CELL_LOC outlo
     func = lookupFluxFunc(FluxTable, method);
   }
 
-  ASSERT1(this == v.getMesh());
-  ASSERT1(this == f.getMesh());
-
   Field2D result(this);
   result.allocate(); // Make sure data allocated
 
+  if (StaggerGrids &&
+      ((v.getLocation() != CELL_CENTRE) || (f.getLocation() != CELL_CENTRE))) {
+    // Staggered differencing
+    throw BoutException("Unhandled staggering");
+  }
+  else {
+    result.setLocation(CELL_CENTRE);
+  }
+
+  ASSERT1(this == v.getMesh());
+  ASSERT1(this == f.getMesh());
```

Comment:
Please could you move these checks to be the first thing in the function call? It's a good idea to bail as early as possible if big assumptions like these aren't met